Transactions/TransactionService.cs

```diff
@@ -28,57 +29,70 @@ public static TransactionService GetTransactionService()
             return _singleton;
         }
 
-        public void InitListByStringArray(string[] data)
+        /*public void InitListByStringArray(string[] data)
         {
             if (data == null) return;
            
             for (int x = 0; x < data.Length / 4; x++)
             {
-                Transaction transaction = new Transaction(data[4 * x], data[x * 4 + 1], data[x * 4 + 2]); //transactionType, amount, transactionTitle
+                Transaction transaction = new Transaction(data[4 * x], data[x * 4 + 1], data[x * 4 + 2], data[x * 4 + 3]); //transactionType, amount, transactionTitle
                 list.Add(transaction);
                 //Console.WriteLine(transaction.Title);
                 //Console.WriteLine(data[4 * x] + x);
             }
-        }
+        }*/
 
         public List<Transaction> GetListWithTitleFiltered(string filter)
         {
             var query = list.Where(oh => oh.Title.ToLower().Contains(filter)).ToList();
-            Console.WriteLine(query.Count);
+            Console.WriteLine("Query count:" + query.Count);
             return query;
 
         }
 
         public void LoadTransactionsListFromTextFile()
         {
+            list.Clear();
             ListItemTransactions[] listItems = new ListItemTransactions[20];
             TextFileReader reader = new TextFileReader();
             string[] data = reader.FetchStringArrayByLocation(System.AppDomain.CurrentDomain.BaseDirectory + @"..\..\Data\Transactions.txt");
-
-            this.InitListByStringArray(data);
-
+            if (data == null) return;
+            for (int x = 0; x < data.Length; x++)
+            {
+                Transaction transaction = JsonConvert.DeserializeObject<Transaction>(data[x]);
+                if(transaction != null)
+                {
+                    list.Add(transaction);
+                }
+                
+            }
```

Comment:
Consider using LINQ for list operations. In general, a regular for should be used only if indexer (i) is really required inside the loop. When it's only for getting an element by the index, it's an indication that it could be rewritten with LINQ.
In this case, the goal is to remap stuff in data to the list of transactions. Then, operators like .Select (which is used to map something) become handy. The whole for could be rewritten like this:
```
var transactions = data.Select(t => JsonConvert.DeserializeObject<Transaction>(t)).Where(t => t != null);
list.AddRange(transactions);
```
This code is shorter and well readable.